plugins/src/main/java/com/google/fhir/gateway/plugin/SmartFhirScope.java

```diff
@@ -132,49 +114,40 @@ private static Set<SmartFhirScopeResourcePermission> extractPermissions(String p
     }
     return permissions;
   }
-}
 
-enum SmartFhirScopeResourcePrincipalContext {
-  USER,
-  PATIENT,
-  SYSTEM;
+  enum PrincipalContext {
+    USER,
+    PATIENT,
+    SYSTEM;
 
-  static SmartFhirScopeResourcePrincipalContext getPrincipalContext(String principal) {
-    switch (principal.toLowerCase()) {
-      case "user":
-        return USER;
-      case "patient":
-        return PATIENT;
-      case "system":
-        return SYSTEM;
-      default:
-        throw new IllegalArgumentException(String.format("Invalid principal %s", principal));
+    static PrincipalContext getPrincipalContext(String principal) {
+      return PrincipalContext.valueOf(principal.toUpperCase());
     }
   }
-}
 
-enum SmartFhirScopeResourcePermission {
-  CREATE,
-  READ,
-  UPDATE,
-  DELETE,
-  SEARCH;
-
-  static SmartFhirScopeResourcePermission getPermission(char permissionCode) {
-    switch (permissionCode) {
-      case 'c':
-        return CREATE;
-      case 'r':
-        return READ;
-      case 'u':
-        return UPDATE;
-      case 'd':
-        return DELETE;
-      case 's':
-        return SEARCH;
-      default:
-        throw new IllegalArgumentException(
-            String.format("Invalid permission code. %s", permissionCode));
+  enum Permission {
```

Comment:
nit: Thanks for relying on `values()` for getting the list of enums here; please add a comment that this order is important (possibly with a reference to FHIR documentation that you mentioned before).